### PR TITLE
Restructure markup & markdown to prepare for multiple markup language…

### PR DESCRIPTION
**Pull request by lunny**
As title.

### CODE REVIEW

File: modules/markup/html.go [hunk 1]
```diff
@@ -0,0 +1,522 @@
...
+
+// Issue name styles
+const (
+	IssueNameStyleNumeric      = "numeric"
+	IssueNameStyleAlphanumeric = "alphanumeric"
+)
+
+var (
+	// NOTE: All below regex matching do not perform any extra validation.
+	// Thus a link is produced even if the user does not exist, the issue does not exist, the commit does not exist, etc.
```
Comment (bkcsoft) [inline modules/markup/html.go]: `even if the linked entity does not exist` would suffice

File: modules/markup/html.go [hunk 2]
```diff
@@ -0,0 +1,522 @@
...
+	// While fast, this is also incorrect and lead to false positives.
```
Comment (bkcsoft) [inline modules/markup/html.go]: Maybe we should document this somewhere else than as a code comment. Or add a `TODO: fix invalid linking issue` or similar.

One thought is having a markdown page in our docs to point our differences in markdown rendering between Gitea, Github, etc. WDYT?

**Review by bkcsoft — COMMENTED** [review]
Few comments, otherwise LGTM

**Comment by lafriks** [comment]
Tests seems to fail

**Comment by lunny** [comment]
@lafriks Yes, I know. I will fix it later.

**Comment by lunny** [comment]
@bkcsoft done. Since this is a refactor PR, I will not change a lot but splitting codes into `markup` and `markdown`. Codes on `markup` will be public by future other doc types. Codes on `markdown` should be the special content for `markdown`.

**Comment by lafriks** [comment]
LGTM

**Comment by lafriks** [comment]
@bkcsoft your LG-TM did not count as it is in review comments

**Comment by bkcsoft** [comment]
LGTM

**Pull request merged**